add js dom menu

alexdefender opened the pull request:
_Complete the form below and remove this line._

### task name

[Demo](https://alexdefender.github.io/js-dom-side-menu/) |
[Code base](https://github.com/alexdefender/js-dom-side-menu)

The code is submitted in a dedicated feature branch.
Please, review.

OleksiyRudenko commented:
@alexdefender You have employed `git-submodule`. Unfortunately submodules aren't tracked by importing repository. That's why we cannot see any code here.

Many also believe submodules are not the best tool to save efforts on importing modules with git.

You may want to employ [git-subrepo](https://github.com/ingydotnet/git-subrepo) instead. It's a kind of git-specific package manager.

alexdefender commented:
Я не использовал submodule. Делал все по инструкции https://github.com/kottans/frontend-2019-homeworks. В локальном git все хорошо, как нужно в папках. А когда запушил на github, получилось вот так. И в следующем регвести так же https://github.com/kottans/frontend-2019-homeworks/pull/221. На каком этапе возникает эта проблема?

OleksiyRudenko commented:
@alexdefender submodules are something that do not apply unless used explicitly.

Can you please create yet another PR with a code in a different folder?
We need log to see what happens and why git submodule engine intervenes. Take a note of your commands and their responses.

Post the log in a that new PR.

One hypothesis is that you have some aliases or standard commands redefined using submodules.

OleksiyRudenko commented:
@alexdefender You fork doesn't contain your code but instead contains a reference to submodules.
![image](https://user-images.githubusercontent.com/8686631/50576479-276ef600-0e1b-11e9-9715-050deb5e0289.png)

The nature of submodules is so you can see the files in your working directory but they aren't tracked by hosting repo.

alexdefender commented:
@OleksiyRudenko не хочу дублировать регвесты, напишу тут. Сделал еще раз по инструкции, результат тот же https://github.com/alexdefender/frontend-2019-homeworks/tree/dom-api-task-1/submissions/alexdefender1

`alex (a-tiny-ls-world) alexdefender $ git checkout master
warning: unable to rmdir submissions/alexdefender/a-tiny-JS-world: Directory not empty
warning: unable to rmdir submissions/alexdefender/js-dom-side-menu: Directory not empty
Switched to branch 'master'
Your branch is ahead of 'origin/master' by 2 commits.
  (use "git push" to publish your local commits)
alex (master) alexdefender $ git status
On branch master
Your branch is ahead of 'origin/master' by 2 commits.
  (use "git push" to publish your local commits)
Untracked files:
  (use "git add <file>..." to include in what will be committed)

	./

nothing added to commit but untracked files present (use "git add" to track)
alex (master) alexdefender $ git pull upstream master
remote: Enumerating objects: 18, done.
remote: Counting objects: 100% (18/18), done.
remote: Total 122 (delta 18), reused 18 (delta 18), pack-reused 104
Receiving objects: 100% (122/122), 235.60 KiB | 336.00 KiB/s, done.
Resolving deltas: 100% (21/21), completed with 10 local objects.
From https://github.com/kottans/frontend-2019-homeworks
 * branch            master     -> FETCH_HEAD
   cf1bef8..3e6c2c9  master     -> upstream/master
Updating cf1bef8..3e6c2c9
Fast-forward
 submissions/Aldegid/memory-pair-game/css/main.css  |  172 +
 .../Aldegid/memory-pair-game/img/001-superhero.svg |    1 +
 .../memory-pair-game/img/002-superhero-1.svg       |    1 +
 .../memory-pair-game/img/003-superhero-2.svg       |    1 +
 .../memory-pair-game/img/004-superhero-3.svg       |    1 +
 .../memory-pair-game/img/005-superhero-4.svg       |    1 +
 .../memory-pair-game/img/006-superhero-5.svg       |    1 +
 .../memory-pair-game/img/007-superhero-6.svg       |    1 +
 .../memory-pair-game/img/008-superhero-7.svg       |    1 +
 .../memory-pair-game/img/014-superhero-13.svg      |    1 +
 .../memory-pair-game/img/020-superhero-19.svg      |    1 +
 .../Aldegid/memory-pair-game/img/texture-2.png     |  Bin 0 -> 28545 bytes
 submissions/Aldegid/memory-pair-game/index.html    |   26 +
 submissions/Aldegid/memory-pair-game/js/script.js  |  117 +
 submissions/IrynaY/memory-pare-game/style.css      |    7 +-
 .../KurosavaAkira/memory-pair_game/.gitignore      |   62 +
 .../KurosavaAkira/memory-pair_game/README.md       |    9 +
 .../build-utils/webpack.development.js             |   36 +
 .../build-utils/webpack.production.js              |   49 +
 .../memory-pair_game/package-lock.json             | 8435 ++++++++++++++++++++
 .../KurosavaAkira/memory-pair_game/package.json    |   38 +
 .../src/components/DeviceSupport/DeviceSupport.js  |   19 +
 .../components/DeviceSupport/DeviceSupport.sass    |    9 +
 .../components/FullScreenVideo/FullScreenVideo.js  |   21 +
 .../FullScreenVideo/FullScreenVideo.sass           |   11 +
 .../memory-pair_game/src/components/HotKeys.js     |   61 +
 .../src/components/MainPage/BattlegroundBoard.js   |   22 +
 .../src/components/MainPage/BattlegroundBoard.sass |   59 +
 .../src/components/MainPage/BattlegroundStats.js   |   17 +
 .../src/components/MainPage/BattlegroundStats.sass |   26 +
 .../src/components/MainPage/CardImages.js          |   26 +
 .../src/components/MainPage/Cards.js               |   75 +
 .../src/components/MainPage/Cards.sass             |   55 +
 .../src/components/MainPage/Countdown.js           |   30 +
 .../src/components/MainPage/Countdown.sass         |   18 +
 .../src/components/MainPage/DialogBox.js           |   56 +
 .../src/components/MainPage/DialogBox.sass         |  111 +
 .../src/components/MainPage/GameLogic.js           |  101 +
 .../src/components/MainPage/GameScene.js           |   13 +
 .../src/components/MainPage/GameScene.sass         |    9 +
 .../src/components/MainPage/MainPage.js            |   19 +
 .../PageTransition/BlackScreenTransition.js        |   26 +
 .../PageTransition/BlackScreenTransition.sass      |   27 +
 .../components/ScorePage/CongratulationSlide.js    |   19 +
 .../components/ScorePage/CongratulationSlide.sass  |   74 +
 .../src/components/ScorePage/ScorePage.js          |   11 +
 .../src/components/ScorePage/ScoreTable.js         |   53 +
 .../src/components/ScorePage/ScoreTable.sass       |   42 +
 .../memory-pair_game/src/components/Settings.js    |    6 +
 .../memory-pair_game/src/components/Sound.js       |   26 +
 .../src/components/StartingPage/DifficultyMenu.js  |   51 +
 .../components/StartingPage/DifficultyMenu.sass    |   49 +
 .../src/components/StartingPage/GameOptions.js     |   90 +
 .../src/components/StartingPage/GameOptions.sass   |  142 +
 .../src/components/StartingPage/IntroVideoMenu.js  |   82 +
 .../components/StartingPage/IntroVideoMenu.sass    |   80 +
 .../src/components/StartingPage/StartingPage.js    |   15 +
 .../memory-pair_game/src/font/capture-it.ttf       |  Bin 0 -> 695372 bytes
 .../memory-pair_game/src/font/p5hatty.ttf          |  Bin 0 -> 20140 bytes
 .../KurosavaAkira/memory-pair_game/src/index.html  |   10 +
 .../KurosavaAkira/memory-pair_game/src/index.js    |   23 +
 .../KurosavaAkira/memory-pair_game/src/index.sass  |   19 +
 .../static/images/favicons/favicon.png             |  Bin 0 -> 885 bytes
 .../memory-pair_game/webpack.config.js             |   23 +
 submissions/igkos/friends-app/index.html           |   39 +
 submissions/igkos/friends-app/index.js             |   95 +
 submissions/igkos/friends-app/style.css            |  114 +
 submissions/justdevway/friends-app/css/main.css    |  471 ++
 submissions/justdevway/friends-app/index.html      |  187 +
 submissions/justdevway/friends-app/js/main.js      |  435 +
 submissions/nazmariam/oop-js/README.md             |    3 +
 submissions/nazmariam/oop-js/css/style.css         |    3 +
 submissions/nazmariam/oop-js/images/1.png          |  Bin 0 -> 12635 bytes
 submissions/nazmariam/oop-js/images/2.png          |  Bin 0 -> 7790 bytes
 submissions/nazmariam/oop-js/images/3.png          |  Bin 0 -> 9210 bytes
 submissions/nazmariam/oop-js/images/Gem Blue.png   |  Bin 0 -> 20160 bytes
 submissions/nazmariam/oop-js/images/Gem Green.png  |  Bin 0 -> 24029 bytes
 submissions/nazmariam/oop-js/images/Gem Orange.png |  Bin 0 -> 17342 bytes
 submissions/nazmariam/oop-js/images/Heart.png      |  Bin 0 -> 6419 bytes
 submissions/nazmariam/oop-js/images/Key.png        |  Bin 0 -> 7455 bytes
 submissions/nazmariam/oop-js/images/Rock.png       |  Bin 0 -> 9434 bytes
 submissions/nazmariam/oop-js/images/Selector.png   |  Bin 0 -> 22430 bytes
 submissions/nazmariam/oop-js/images/Star.png       |  Bin 0 -> 13735 bytes
 submissions/nazmariam/oop-js/images/char-boy.png   |  Bin 0 -> 19948 bytes
 .../nazmariam/oop-js/images/char-cat-girl.png      |  Bin 0 -> 8553 bytes
 .../nazmariam/oop-js/images/char-horn-girl.png     |  Bin 0 -> 8724 bytes
 .../nazmariam/oop-js/images/char-pink-girl.png     |  Bin 0 -> 8868 bytes
 .../nazmariam/oop-js/images/char-princess-girl.png |  Bin 0 -> 9735 bytes
 .../nazmariam/oop-js/images/enemy-bug (copy).png   |  Bin 0 -> 12635 bytes
 submissions/nazmariam/oop-js/images/enemy-bug.png  |  Bin 0 -> 11513 bytes
 .../nazmariam/oop-js/images/grass-block.png        |  Bin 0 -> 20489 bytes
 .../nazmariam/oop-js/images/stone-block.png        |  Bin 0 -> 22172 bytes
 .../nazmariam/oop-js/images/water-block.png        |  Bin 0 -> 20489 bytes
 submissions/nazmariam/oop-js/index.html            |   13 +
 submissions/nazmariam/oop-js/js/app.js             |  112 +
 submissions/nazmariam/oop-js/js/engine.js          |  185 +
 submissions/nazmariam/oop-js/js/resources.js       |  110 +
 97 files changed, 12351 insertions(+), 3 deletions(-)
 create mode 100644 submissions/Aldegid/memory-pair-game/css/main.css
 create mode 100644 submissions/Aldegid/memory-pair-game/img/001-superhero.svg
 create mode 100644 submissions/Aldegid/memory-pair-game/img/002-superhero-1.svg
 create mode 100644 submissions/Aldegid/memory-pair-game/img/003-superhero-2.svg
 create mode 100644 submissions/Aldegid/memory-pair-game/img/004-superhero-3.svg
 create mode 100644 submissions/Aldegid/memory-pair-game/img/005-superhero-4.svg
 create mode 100644 submissions/Aldegid/memory-pair-game/img/006-superhero-5.svg
 create mode 100644 submissions/Aldegid/memory-pair-game/img/007-superhero-6.svg
 create mode 100644 submissions/Aldegid/memory-pair-game/img/008-superhero-7.svg
 create mode 100644 submissions/Aldegid/memory-pair-game/img/014-superhero-13.svg
 create mode 100644 submissions/Aldegid/memory-pair-game/img/020-superhero-19.svg
 create mode 100644 submissions/Aldegid/memory-pair-game/img/texture-2.png
 create mode 100644 submissions/Aldegid/memory-pair-game/index.html
 create mode 100644 submissions/Aldegid/memory-pair-game/js/script.js
 create mode 100644 submissions/KurosavaAkira/memory-pair_game/.gitignore
 create mode 100644 submissions/KurosavaAkira/memory-pair_game/README.md
 create mode 100644 submissions/KurosavaAkira/memory-pair_game/build-utils/webpack.development.js
 create mode 100644 submissions/KurosavaAkira/memory-pair_game/build-utils/webpack.production.js
 create mode 100644 submissions/KurosavaAkira/memory-pair_game/package-lock.json
 create mode 100644 submissions/KurosavaAkira/memory-pair_game/package.json
 create mode 100644 submissions/KurosavaAkira/memory-pair_game/src/components/DeviceSupport/DeviceSupport.js
 create mode 100644 submissions/KurosavaAkira/memory-pair_game/src/components/DeviceSupport/DeviceSupport.sass
 create mode 100644 submissions/KurosavaAkira/memory-pair_game/src/components/FullScreenVideo/FullScreenVideo.js
 create mode 100644 submissions/KurosavaAkira/memory-pair_game/src/components/FullScreenVideo/FullScreenVideo.sass
 create mode 100644 submissions/KurosavaAkira/memory-pair_game/src/components/HotKeys.js
 create mode 100644 submissions/KurosavaAkira/memory-pair_game/src/components/MainPage/BattlegroundBoard.js
 create mode 100644 submissions/KurosavaAkira/memory-pair_game/src/components/MainPage/BattlegroundBoard.sass
 create mode 100644 submissions/KurosavaAkira/memory-pair_game/src/components/MainPage/BattlegroundStats.js
 create mode 100644 submissions/KurosavaAkira/memory-pair_game/src/components/MainPage/BattlegroundStats.sass
 create mode 100644 submissions/KurosavaAkira/memory-pair_game/src/components/MainPage/CardImages.js
 create mode 100644 submissions/KurosavaAkira/memory-pair_game/src/components/MainPage/Cards.js
 create mode 100644 submissions/KurosavaAkira/memory-pair_game/src/components/MainPage/Cards.sass
 create mode 100644 submissions/KurosavaAkira/memory-pair_game/src/components/MainPage/Countdown.js
 create mode 100644 submissions/KurosavaAkira/memory-pair_game/src/components/MainPage/Countdown.sass
 create mode 100644 submissions/KurosavaAkira/memory-pair_game/src/components/MainPage/DialogBox.js
 create mode 100644 submissions/KurosavaAkira/memory-pair_game/src/components/MainPage/DialogBox.sass
 create mode 100644 submissions/KurosavaAkira/memory-pair_game/src/components/MainPage/GameLogic.js
 create mode 100644 submissions/KurosavaAkira/memory-pair_game/src/components/MainPage/GameScene.js
 create mode 100644 submissions/KurosavaAkira/memory-pair_game/src/components/MainPage/GameScene.sass
 create mode 100644 submissions/KurosavaAkira/memory-pair_game/src/components/MainPage/MainPage.js
 create mode 100644 submissions/KurosavaAkira/memory-pair_game/src/components/PageTransition/BlackScreenTransition.js
 create mode 100644 submissions/KurosavaAkira/memory-pair_game/src/components/PageTransition/BlackScreenTransition.sass
 create mode 100644 submissions/KurosavaAkira/memory-pair_game/src/components/ScorePage/CongratulationSlide.js
 create mode 100644 submissions/KurosavaAkira/memory-pair_game/src/components/ScorePage/CongratulationSlide.sass
 create mode 100644 submissions/KurosavaAkira/memory-pair_game/src/components/ScorePage/ScorePage.js
 create mode 100644 submissions/KurosavaAkira/memory-pair_game/src/components/ScorePage/ScoreTable.js
 create mode 100644 submissions/KurosavaAkira/memory-pair_game/src/components/ScorePage/ScoreTable.sass
 create mode 100644 submissions/KurosavaAkira/memory-pair_game/src/components/Settings.js
 create mode 100644 submissions/KurosavaAkira/memory-pair_game/src/components/Sound.js
 create mode 100644 submissions/KurosavaAkira/memory-pair_game/src/components/StartingPage/DifficultyMenu.js
 create mode 100644 submissions/KurosavaAkira/memory-pair_game/src/components/StartingPage/DifficultyMenu.sass
 create mode 100644 submissions/KurosavaAkira/memory-pair_game/src/components/StartingPage/GameOptions.js
 create mode 100644 submissions/KurosavaAkira/memory-pair_game/src/components/StartingPage/GameOptions.sass
 create mode 100644 submissions/KurosavaAkira/memory-pair_game/src/components/StartingPage/IntroVideoMenu.js
 create mode 100644 submissions/KurosavaAkira/memory-pair_game/src/components/StartingPage/IntroVideoMenu.sass
 create mode 100644 submissions/KurosavaAkira/memory-pair_game/src/components/StartingPage/StartingPage.js
 create mode 100644 submissions/KurosavaAkira/memory-pair_game/src/font/capture-it.ttf
 create mode 100644 submissions/KurosavaAkira/memory-pair_game/src/font/p5hatty.ttf
 create mode 100644 submissions/KurosavaAkira/memory-pair_game/src/index.html
 create mode 100644 submissions/KurosavaAkira/memory-pair_game/src/index.js
 create mode 100644 submissions/KurosavaAkira/memory-pair_game/src/index.sass
 create mode 100644 submissions/KurosavaAkira/memory-pair_game/static/images/favicons/favicon.png
 create mode 100644 submissions/KurosavaAkira/memory-pair_game/webpack.config.js
 create mode 100644 submissions/igkos/friends-app/index.html
 create mode 100644 submissions/igkos/friends-app/index.js
 create mode 100644 submissions/igkos/friends-app/style.css
 create mode 100644 submissions/justdevway/friends-app/css/main.css
 create mode 100644 submissions/justdevway/friends-app/index.html
 create mode 100644 submissions/justdevway/friends-app/js/main.js
 create mode 100644 submissions/nazmariam/oop-js/README.md
 create mode 100644 submissions/nazmariam/oop-js/css/style.css
 create mode 100755 submissions/nazmariam/oop-js/images/1.png
 create mode 100755 submissions/nazmariam/oop-js/images/2.png
 create mode 100755 submissions/nazmariam/oop-js/images/3.png
 create mode 100755 submissions/nazmariam/oop-js/images/Gem Blue.png
 create mode 100755 submissions/nazmariam/oop-js/images/Gem Green.png
 create mode 100755 submissions/nazmariam/oop-js/images/Gem Orange.png
 create mode 100755 submissions/nazmariam/oop-js/images/Heart.png
 create mode 100755 submissions/nazmariam/oop-js/images/Key.png
 create mode 100755 submissions/nazmariam/oop-js/images/Rock.png
 create mode 100755 submissions/nazmariam/oop-js/images/Selector.png
 create mode 100755 submissions/nazmariam/oop-js/images/Star.png
 create mode 100644 submissions/nazmariam/oop-js/images/char-boy.png
 create mode 100755 submissions/nazmariam/oop-js/images/char-cat-girl.png
 create mode 100755 submissions/nazmariam/oop-js/images/char-horn-girl.png
 create mode 100755 submissions/nazmariam/oop-js/images/char-pink-girl.png
 create mode 100755 submissions/nazmariam/oop-js/images/char-princess-girl.png
 create mode 100755 submissions/nazmariam/oop-js/images/enemy-bug (copy).png
 create mode 100644 submissions/nazmariam/oop-js/images/enemy-bug.png
 create mode 100755 submissions/nazmariam/oop-js/images/grass-block.png
 create mode 100755 submissions/nazmariam/oop-js/images/stone-block.png
 create mode 100644 submissions/nazmariam/oop-js/images/water-block.png
 create mode 100644 submissions/nazmariam/oop-js/index.html
 create mode 100644 submissions/nazmariam/oop-js/js/app.js
 create mode 100644 submissions/nazmariam/oop-js/js/engine.js
 create mode 100644 submissions/nazmariam/oop-js/js/resources.js
alex (master) alexdefender $ git checkout -b dom-api-task-1
Switched to a new branch 'dom-api-task-1'
alex (dom-api-task-1) alexdefender $ git commit -m "add dom-api-task-1"
On branch dom-api-task-1
Untracked files:
	./
	../alexdefender1/

nothing added to commit but untracked files present
alex (dom-api-task-1) alexdefender $ git push --set-upstream origin dom-api-task-1
Username for 'https://github.com': alex.iv1102@gmail.com
Password for 'https://alex.iv1102@gmail.com@github.com': 
Counting objects: 122, done.
Delta compression using up to 2 threads.
Compressing objects: 100% (97/97), done.
Writing objects: 100% (122/122), 235.45 KiB | 0 bytes/s, done.
Total 122 (delta 20), reused 119 (delta 18)
remote: Resolving deltas: 100% (20/20), completed with 9 local objects.
remote: 
remote: Create a pull request for 'dom-api-task-1' on GitHub by visiting:
remote:      https://github.com/alexdefender/frontend-2019-homeworks/pull/new/dom-api-task-1
remote: 
To https://github.com/alexdefender/frontend-2019-homeworks.git
 * [new branch]      dom-api-task-1 -> dom-api-task-1
Branch dom-api-task-1 set up to track remote branch dom-api-task-1 from origin.
alex (dom-api-task-1) alexdefender $ git status
On branch dom-api-task-1
Your branch is up-to-date with 'origin/dom-api-task-1'.
Untracked files:
  (use "git add <file>..." to include in what will be committed)

	./
	../alexdefender1/

nothing added to commit but untracked files present (use "git add" to track)
alex (dom-api-task-1) alexdefender $ git add .
alex (dom-api-task-1 +) alexdefender $ git status
On branch dom-api-task-1
Your branch is up-to-date with 'origin/dom-api-task-1'.
Changes to be committed:
  (use "git reset HEAD <file>..." to unstage)

	new file:   a-tiny-JS-world
	new file:   js-dom-side-menu

Untracked files:
  (use "git add <file>..." to include in what will be committed)

	../alexdefender1/

alex (dom-api-task-1 +) alexdefender $ ls
a-tiny-JS-world  js-dom-side-menu
alex (dom-api-task-1 +) alexdefender $ cd ..
alex (dom-api-task-1 +) submissions $ git status
On branch dom-api-task-1
Your branch is up-to-date with 'origin/dom-api-task-1'.
Changes to be committed:
  (use "git reset HEAD <file>..." to unstage)

	new file:   alexdefender/a-tiny-JS-world
	new file:   alexdefender/js-dom-side-menu

Untracked files:
  (use "git add <file>..." to include in what will be committed)

	alexdefender1/

alex (dom-api-task-1 +) submissions $ git add .
alex (dom-api-task-1 +) submissions $ git status
On branch dom-api-task-1
Your branch is up-to-date with 'origin/dom-api-task-1'.
Changes to be committed:
  (use "git reset HEAD <file>..." to unstage)

	new file:   alexdefender/a-tiny-JS-world
	new file:   alexdefender/js-dom-side-menu
	new file:   alexdefender1/js-dom-side-menu

alex (dom-api-task-1 +) submissions $ git commit -m "add dom-api-task-1"[dom-api-task-1 8c75cd4] add dom-api-task-1
 3 files changed, 3 insertions(+)
 create mode 160000 submissions/alexdefender/a-tiny-JS-world
 create mode 160000 submissions/alexdefender/js-dom-side-menu
 create mode 160000 submissions/alexdefender1/js-dom-side-menu
alex (dom-api-task-1) submissions $ git log --oneline
8c75cd4 add dom-api-task-1
3e6c2c9 Add memory-pair_game (#170)
dc22a42 Friends-app (#182)
09a2b48 Friends App (#186)
3f2686f add memory pair game task (#197)
bc9e364 Happy new year! (#217)
cf1bef8 update app.js (HNY) (#216)
91eef86 oop tiny js world (#206)
3bf769e add js-oop-frogger task (#200)
bfd0220 Dom api task (#190)
c65308b Merge pull request #133 from bugagashinka/dom-api-task
547cb7d Merge pull request #180 from o-msh/friends-app
eed97db js-pre-oop (#187)
80cd78b Add a tiny JS world OOP version (#181)
ff24cc0 Finished with oop exercise (#171)
e3bf849 A tiny js world (#198)
5cb9531 Fix css, layouts
da456ec Memory pair game (#196)
bb00ed5 Fix
a5e2188 Oop exercise (#130)
7166982 Memory pair game (#167)
0141f15 add memory-pair-game (#199)
f6b675c Add first homework js-dom (#131)
alex (dom-api-task-1) submissions $ git push --set-upstream origin dom-api-task-1
Username for 'https://github.com': alex.iv1102@gmail.com
Password for 'https://alex.iv1102@gmail.com@github.com': 
Counting objects: 5, done.
Delta compression using up to 2 threads.
Compressing objects: 100% (4/4), done.
Writing objects: 100% (5/5), 499 bytes | 0 bytes/s, done.
Total 5 (delta 2), reused 0 (delta 0)
remote: Resolving deltas: 100% (2/2), completed with 2 local objects.
To https://github.com/alexdefender/frontend-2019-homeworks.git
   3e6c2c9..8c75cd4  dom-api-task-1 -> dom-api-task-1
Branch dom-api-task-1 set up to track remote branch dom-api-task-1 from origin.
alex (dom-api-task-1) submissions $ git status
On branch dom-api-task-1
Your branch is up-to-date with 'origin/dom-api-task-1'.
nothing to commit, working directory clean
alex (dom-api-task-1) submissions $ ^C
alex (dom-api-task-1) submissions $ 
`

alexdefender commented:
Отдельно эти задания отображаются нормально
https://github.com/alexdefender/a-tiny-JS-world/tree/gh-pages
https://github.com/alexdefender/js-dom-side-menu

А почему-то в frontend-2019-homeworks
https://github.com/alexdefender/frontend-2019-homeworks/tree/dom-api-task-1/submissions/alexdefender1
как субмодули.

alexdefender commented:
Разобрался в чем проблема, сейчас все пофиксю.

alexdefender commented:
Вроде все получилось. Проверьте, пожалуйста.

A-Ostrovnyy commented on submissions/alexdefender/js-dom-side-menu/css/style.css:
```diff
@@ -0,0 +1,57 @@
+    padding: 20px 20px 20px 20px;
+}
+
+.content img {
```
Avoid styling by tag add custom styles on class.
For `<body>` leave but for other tags it is better to assign them a class and style them according to the class.
When a project is large and the team is working on it, it avoids the constant dropping of styles for the tag in different places. In addition, the class can be applied to different tags.

A-Ostrovnyy commented on submissions/alexdefender/js-dom-side-menu/css/style.css:
```diff
@@ -0,0 +1,57 @@
+@media screen and (min-width: 300px) {
```
Media rules goes after main styles code block.

A-Ostrovnyy commented on submissions/alexdefender/js-dom-side-menu/js/js.js:
```diff
@@ -0,0 +1,92 @@
+}]
+
+var menuItem1 = document.querySelector('.menu-item-1');
+menuItem1.addEventListener('click', function () {
```
Use Event delegation [https://learn.javascript.ru/event-delegation](https://learn.javascript.ru/event-delegation) and  [https://davidwalsh.name/event-delegate](https://davidwalsh.name/event-delegate).
In this case you don't need so many unique class.

A-Ostrovnyy commented on submissions/alexdefender/js-dom-side-menu/index.html:
```diff
@@ -0,0 +1,28 @@
+        <div class="menu">
+            <ul class="menu-list">
+                <li class="menu-title">Оттенки зеленого цвета:</li>
+                <li class="menu-item-1">Зеленый цвет 01</li>
```
If you will use event delegation(see links below in the .js) you need just only one classname `.menu-item`.

alexdefender replied:
Не могу понять, как будет выводиться нужный `contentGoods` с `contentArr`, без использования разных `menu-item` или `id` в `li`?

A-Ostrovnyy replied:
Use Event delegation [https://learn.javascript.ru/event-delegation](https://learn.javascript.ru/event-delegation) and  [https://davidwalsh.name/event-delegate](https://davidwalsh.name/event-delegate).
In this case you don't need so many unique class.

A-Ostrovnyy commented on submissions/alexdefender/js-dom-side-menu/js/js.js:
```diff
@@ -0,0 +1,92 @@
+  chooseContent.appendChild(newDiv);
+});
+
+function applyStyleCss(element) {
```
Don't apply style by js, this is bad practice, create style block in css with unique classname and add him by js.

A-Ostrovnyy commented on submissions/alexdefender/js-dom-side-menu/js/js.js:
```diff
@@ -0,0 +1,92 @@
+    element.removeChild(element.firstChild);
+  }
+
+  let newDiv = document.createElement('div');
```
This block of code duplicated a lot of times. Create one independent abstract method and call him for every `.menu-item`. Don't duplicate identical method.

A-Ostrovnyy commented on submissions/alexdefender/js-dom-side-menu/index.html:
```diff
@@ -0,0 +1,28 @@
+                <li class="menu-item-5">Зеленый цвет 05</li>
+            </ul>
+        </div>
+        <div class="content">  
```
For enter point for js better use `id` (`class` for style).

alexdefender commented:
@A-Ostrovnyy @IgorKurkov please, review.

A-Ostrovnyy commented:
Good work!
